frontend/app/services/lookup-service.server.ts

```diff
@@ -0,0 +1,65 @@
+import { z } from 'zod';
+
+import { getLogger } from '~/utils/logging.server';
+
+const logger = getLogger('user-service.server');
+
+export const preferredLanguageSchema = z.object({
+  id: z.string().optional(),
+  nameEn: z.string().optional(),
+  nameFr: z.string().optional(),
+});
+
+export type PreferredLanguageInfo = z.infer<typeof preferredLanguageSchema>;
+
+export interface LookupServiceDependencies {
+  readonly env: {
+    readonly INTEROP_API_BASE_URI: string;
+  };
+}
+
+export function getLookupService({ env }: LookupServiceDependencies) {
+  const { INTEROP_API_BASE_URI } = env;
+
+  async function getAllPreferredLanguages() {
+    const url = `${INTEROP_API_BASE_URI}/lookups/preferred-languages/`;
+    const response = await fetch(url);
+    const body = await response.json();
+    const preferredLanguageSchemaList = z.array(preferredLanguageSchema);
+    if (response.ok) return {initalValues: preferredLanguageSchemaList.parse(body) };
```

Comment:
Get, parse and return parsed JSON if response is OK.
```suggestion
    if (response.ok) return preferredLanguageSchemaList.parse(await response.json());
```

Reply:
Fixed: 72170e807b3f47c361f34e4bd4d9036444475779